components/kyma-environment-broker/internal/process/provisioning/input/input.go

```diff
@@ -127,7 +136,15 @@ func (r *RuntimeInput) applyProvisioningParameters() error {
 }
 
 func (r *RuntimeInput) disableNotSelectedComponents() error {
-	toDisable := r.optionalComponentsService.ComputeComponentsToDisable(r.provisioningParameters.OptionalComponentsToInstall)
+	r.mutex.Lock("components")
```

Comment:
minor: For consistency across other methods, you could name that channel `EnableComponent`